fetch/api/abort/serviceworker-intercepted.https.html

```diff
@@ -100,6 +103,92 @@
     await promise_rejects_dom(t, "AbortError", w.DOMException, reader.read());
     await promise_rejects_dom(t, "AbortError", w.DOMException, reader.closed);
   }, "Stream errors once aborted.");
+
+  promise_test(async t => {
+    const scope = SCOPE + "?q=aborted-with-abort-reason";
+    await setupRegistration(t, scope, '../resources/sw-intercept.js');
+    const iframe = await with_iframe(scope);
+    add_completion_callback(_ => iframe.remove());
+    const w = iframe.contentWindow;
+
+    const controller = new w.AbortController();
+    const signal = controller.signal;
+
+    const fetchPromise = await w.fetch('data.json', { signal });
```

Comment:
The test does not produce the required failure messages in the [tests](https://wpt.fyi/results/fetch/api/abort/serviceworker-intercepted.https.html?label=master&label=experimental&aligned&view=subtest&q=serviceworker-intercepted.https.html)

I think we await here should be removed. 
The above code will return a response object  and not the promise which we need to pass it to promise_rejects_exactly. 
Kindly check, I am not a JS expert :)

Reply:
That seems correct @MayyaSunil. Could you create a PR?

Reply:
Sure @annevk. I will submit the fix in mozilla's repo and  let the sync bot do the job? Is it acceptable to you?

Reply:
Yeah that works. It might be worth having another test like this whereby we do wait for the response and then check that `response.body` was errored with the correct exception if that doesn't exist already.

Reply:
[Bug 1798921](https://bugzilla.mozilla.org/show_bug.cgi?id=1798921) is created to track this

Reply:
If the below patch is  fine, I will submit it to our repo.

```diff
--- a/testing/web-platform/tests/fetch/api/abort/serviceworker-intercepted.https.html
+++ b/testing/web-platform/tests/fetch/api/abort/serviceworker-intercepted.https.html
@@ -114,13 +114,31 @@
     const controller = new w.AbortController();
     const signal = controller.signal;
 
-    const fetchPromise = await w.fetch('data.json', { signal });
+    const fetchPromise = w.fetch('data.json', { signal });
 
     controller.abort(error1);
 
     await promise_rejects_exactly(t, error1, fetchPromise);
   }, "fetch() rejects with abort reason");
 
+
+  promise_test(async t => {
+    const scope = SCOPE + "?q=aborted-with-abort-reason-in-body";
+    await setupRegistration(t, scope, '../resources/sw-intercept.js');
+    const iframe = await with_iframe(scope);
+    add_completion_callback(_ => iframe.remove());
+    const w = iframe.contentWindow;
+
+    const controller = new w.AbortController();
+    const signal = controller.signal;
+
+    const fetchResponse = await w.fetch('data.json', { signal });
+    const bodyPromise  = fetchResponse.body.getReader().read();
+    controller.abort(error1);
+
+    await promise_rejects_exactly(t, error1, bodyPromise);
+    }, "fetch() response body has abort reason");

```

Reply:
That looks correct, yeah. Thanks @MayyaSunil!